kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSPropertyDeclarationJavaImpl.kt

```diff
@@ -0,0 +1,120 @@
+package com.google.devtools.ksp.impl.symbol.kotlin
+
+import com.google.devtools.ksp.KSObjectCache
+import com.google.devtools.ksp.symbol.KSAnnotation
+import com.google.devtools.ksp.symbol.KSDeclaration
+import com.google.devtools.ksp.symbol.KSFile
+import com.google.devtools.ksp.symbol.KSName
+import com.google.devtools.ksp.symbol.KSNode
+import com.google.devtools.ksp.symbol.KSPropertyDeclaration
+import com.google.devtools.ksp.symbol.KSPropertyGetter
+import com.google.devtools.ksp.symbol.KSPropertySetter
+import com.google.devtools.ksp.symbol.KSType
+import com.google.devtools.ksp.symbol.KSTypeParameter
+import com.google.devtools.ksp.symbol.KSTypeReference
+import com.google.devtools.ksp.symbol.KSVisitor
+import com.google.devtools.ksp.symbol.Location
+import com.google.devtools.ksp.symbol.Modifier
+import com.google.devtools.ksp.symbol.Origin
+import com.google.devtools.ksp.toKSModifiers
+import org.jetbrains.kotlin.analysis.api.symbols.KtJavaFieldSymbol
+import org.jetbrains.kotlin.psi.KtModifierListOwner
+import org.jetbrains.kotlin.utils.addToStdlib.safeAs
+
+class KSPropertyDeclarationJavaImpl private constructor(
+    private val ktJavaFieldSymbol: KtJavaFieldSymbol
```

Comment:
Just asking :) What's the plan for `KtSyntheticJavaPropertySymbol`? `KSPropertyDeclarationSyntheticJavaPropertyImpl` or `KSSyntheticJavaPropertyDeclarationImpl`? Maybe this one should emphasize java "field" somehow?

Reply:
This is essentially just field, the name `property` is quite confusing tbf, but since KSP models everything in the Kotlin way, we just use `property` to describe java fields.